engine.go

```diff
@@ -539,6 +545,57 @@ func (t *T) shouldLog() bool {
 	return t.rawLog != nil || t.tbLog
 }
 
+// Context returns a context.Context associated with the test.
+// It is valid only for the duration of the rapid check.
+func (t *T) Context() context.Context {
+	// Fast path: no need to lock if the context is already set.
+	t.mu.RLock()
+	if t.ctx != nil {
+		t.mu.RUnlock()
+		return t.ctx
```

Comment:
This is racy: after the unlock `t.ctx` may already be `nil`.

Reply:
🤦  derp. I had this right the first time around and decided to clean up some local variables for fun.

Reply:
Fix pushed